### ISSUE TITLE
Character window_background and window hide transition

### ISSUE THREAD

**Issue by ticlock** [renpy/renpy#4527]
A character-specified `window_background` is not used in window hide transitions. `window hide` uses `style.window.background` rather than `window_background` from the last character. 

As a result, the character-specified dialogue textbox is removed, the default dialogue textbox appears and only then the hide transition takes place. 

For example, this happens using choice menu (with `window auto`)
```
define e = Character("Eileen",  window_background = "window_background_eileen")
```
```
    e "Line 1"
    menu:
        ...
```

Using an empty style.window.background or removing the transition can help to avoid the appearance of the default textbox, but then there is no visual hide transition for the dialogue textbox.

Possibly, It makes sense to store the last window_backgound and use it for hide statements.

**Comment by renpytom** [comment]
This is correct, and by design - at least for the moment. I'll leave this open as a feature, but for now this is how the empty window works - it doesn't consider what's shown before.

**Comment by ewolf201** [comment]
Any updates to this features? it's been while and just now encountered this problem in a project of mine and it driving me nuts ngl.

**Comment by renpytom** [comment]
No, there  hasn't been any effort put into this. It would be a major change - I've properly flagged it as one.
